src/main/java/com/hazelcast/jdbc/JdbcConnection.java

```
@@ -0,0 +1,352 @@
+import java.sql.SQLWarning;
+import java.sql.SQLXML;
+import java.sql.Savepoint;
+import java.sql.Statement;
+import java.sql.Struct;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.Executor;
+
+class JdbcConnection implements Connection {
+
+    private final HazelcastInstance client;
+
+    /** Close the gateway. */
+    private boolean closed;
+
+    JdbcConnection(HazelcastInstance client) {
+        this.client = client;
+    }
+
+    @Override
+    public Statement createStatement() throws SQLException {
+        checkClosed();
+        return new HazelcastJdbcStatement(client);
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String sql) throws SQLException {
+        checkClosed();
+        return new HazelcastJdbcPreparedStatement(sql, client);
+    }
+
+    @Override
+    public CallableStatement prepareCall(String sql) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("CallableStatement not supported");
+    }
+
+    @Override
+    public String nativeSQL(String sql) throws SQLException {
+        checkClosed();
+        return sql;
+    }
+
+    @Override
+    public void setAutoCommit(boolean autoCommit) throws SQLException {
+        checkClosed();
+    }
+
+    @Override
+    public boolean getAutoCommit() throws SQLException {
+        checkClosed();
+        return false;
+    }
+
+    @Override
+    public void commit() throws SQLException {
+        checkClosed();
+    }
+
+    @Override
+    public void rollback() throws SQLException {
+        checkClosed();
+    }
+
+    @Override
+    public void close() throws SQLException {
+        if (!isClosed()) {
+            client.shutdown();
+            closed = true;
+        }
+    }
+
+    @Override
+    public boolean isClosed() throws SQLException {
+        return closed;
+    }
+
+    @Override
+    public DatabaseMetaData getMetaData() throws SQLException {
+        checkClosed();
+        return null;
```

Until the metadata is implemented, it is better to throw an exception here instead of returning `null`, to have predictable behavior.